benchmarking/simulation/simulation.py

```diff
@@ -33,6 +34,14 @@ def __init__(
         self.team_sets = []
 
     def run(self, num_runs: int) -> SimulationArtifact:
+        if self.settings.cache_key:
+            cache = SimulationCache(self.settings.cache_key)
+            if cache.exists():
+                self.team_sets = cache.get_teams()
+                self.run_times = cache.get_runtimes()
+                # TODO: If len(self.team_sets) < num_runs, run the remaining runs
```

Comment:
Make a separate ticket for this, I think we include this in the start/stop functionality